lib/semantics/check-omp-structure.h

```diff
@@ -146,16 +145,13 @@ class OmpStructureChecker : public virtual BaseChecker {
     CHECK(!ompContext_.empty());
     return ompContext_.back();
   }
+  void ClearContext() {
+    CHECK(!ompContext_.empty());
+    ompContext_.back() = {};
```

Comment:
I think this will set the `directive` data member to `OmpDirective::PARALLEL` because it will be zero-initialized and that is the zero value of `OmpDirective`. That doesn't seem like a good default value.

It would be better to make `directive` have type `std::optional<OmpDirective>`. Then access it as `GetContext().directive.value()` so if it hasn't been set you'll get a runtime error and not `OmpDirective::PARALLEL`. Then maybe put that into a member function so you can get the directive with `GetContextDirective()`.

Reply:
By the way, this problem already exists. The version of `PushContext()` that doesn't set the directive also leaves it in a questionable state until it is set.